src/wallet/injected/wallet-connection-sender.ts

```diff
@@ -0,0 +1,54 @@
+import {
+    RequestSignTransactionsOptions,
+    SignInOptions,
+    WalletConnection,
+} from "../wallet-connection";
+
+import { Near } from '../../near';
+import { Account } from '../../account'
+import { SenderWallet } from "./wallet-injected-interface";
+
+export class WalletConnectionSender extends WalletConnection {
+    private senderWallet: SenderWallet;
+
+
+    constructor(near: Near, appKeyPrefix: string | null, walletName: string) {
+        console.warn("This implementation is temporary, regular Injected Wallet should be used instead.");
+        super(near, appKeyPrefix);
+        this.senderWallet = window[walletName] as SenderWallet;
+    }
+
+    async requestSignTransactions({ transactions, meta, callbackUrl }: RequestSignTransactionsOptions): Promise<void> {
+        this.senderWallet.requestSignTransactions({ transactions })
+            .then((res) => {
+                console.log('requestSignTransactions res:', res);
+                return res;
+            });
+    }
+    async requestSignIn({ contractId, methodNames, successUrl, failureUrl }: SignInOptions): Promise<void> {
+        const { accessKey } = await this.senderWallet.requestSignIn({
+            contractId: contractId,
+        });
+        //TODO: why do we need accessKey here?
+        console.log("New function call key added:", accessKey);
+    }
+    isSignedIn(): boolean {
+        return this.senderWallet.isSignedIn();
+    }
+    getAccountId(): string {
+        return this.senderWallet.getAccountId();
+    }
+    signOut(): boolean {
+        this.senderWallet.signOut().then((res) => {
+            if (res.result !== "success") {
+                throw new Error("Failed to sign out");
+            } else {
+                console.log("Signed out");
+            }
+        });
+        return true;
+    }
+    account(): Account {
+        throw new Error('Method not implemented.');
```

Comment:
I also suggest to implement this in Sender, but it may also require some change in `near-api-js`.

Reply:
In the current design, the wallet should not return the Account object. First of all, it is not possible for Redirect (NEAR) Wallet, so the experience will not be consistent.

Reply:
why? I thought we use reuse the [`account()`](https://github.com/near/near-api-js/blob/e92725352a5ad33db7ba97f4d181da44af88267f/src/wallet-account.ts#L264-L272) method in `WalletConnection`. but I may misunderstand the latest design